### PR TITLE
Improve performance of LongObjectPagedHashMap#removeAndAdd and ObjectObjectPagedHashMap#removeAndAdd

### PR DESCRIPTION
**Pull request by iverase**
Similarly to what we did in https://github.com/elastic/elasticsearch/pull/114199, lets remove some unnecessary manipulation of the keys in the method removeAndAdd.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/util/LongObjectPagedHashMap.java [hunk 1]
```diff
@@ -94,9 +95,6 @@ public T remove(long key) {
     }
 
     private T set(long key, T value) {
-        if (value == null) {
-            throw new IllegalArgumentException("Null values are not supported");
-        }
```
Comment (iverase) [inline server/src/main/java/org/elasticsearch/common/util/LongObjectPagedHashMap.java:99]: Let's just assert it and don't waste time checking for it

File: server/src/main/java/org/elasticsearch/common/util/LongObjectPagedHashMap.java [hunk 2]
```diff
@@ -181,9 +179,19 @@ protected boolean used(long bucket) {
...
         final T value = values.getAndSet(index, null);
-        --size;
-        final T removed = set(key, value);
-        assert removed == null;
+        reset(key, value);
+    }
+
+    private void reset(long key, T value) {
+        for (long i = slot(hash(key), mask);; i = nextSlot(i, mask)) {
+            final T previous = values.get(i);
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/common/util/LongObjectPagedHashMap.java:187]: Let's make values a local variable again? :)

**Review by original-brownbear — APPROVED** [review]
LGTM, maybe make values a local again for extra fun :)

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**